ads/google/a4a/test/test-utils.js

```diff
@@ -179,72 +212,80 @@ describe('Google A4A utils', () => {
-          /s=a4a/,
-          /&c=[0-9]+/,
-          /&slotId=null/,
-          /&qqid\.null=[a-zA-Z_]+/,
-          new RegExp(`&met\\.a4a\\.null=${metricName}\\.-?[0-9]+`),
-          /&dt=-?[0-9]+/,
-          /e\.null=00000001%2C0000002/,
-          /rls=\$internalRuntimeVersion\$/,
-          /adt.null=(doubleclick|adsense)/,
-        ];
-        getRegExps('iniLoadCsi').forEach(regExp => {
-          expect(iniLoadCsiRequest).to.match(regExp);
-        });
-        getRegExps('renderStartCsi').forEach(regExp => {
-          expect(renderStartCsiRequest).to.match(regExp);
-        });
-        // Need to remove this request as it will vary in test execution.
-        delete config.requests.iniLoadCsi;
-        delete config.requests.renderStartCsi;
-        expect(config).to.deep.equal({
-          transport: {beacon: false, xhrpost: false},
-          requests: {
-            visibility1: url[0],
-            visibility2: url[1],
-          },
-          triggers: {
-            continuousVisible: {
-              on: 'visible',
-              request: ['visibility1', 'visibility2'],
-              visibilitySpec: {
-                selector: 'amp-ad',
-                selectionMethod: 'closest',
-                visiblePercentageMin: 50,
-                continuousTimeMin: 1000,
-              },
-            },
-            continuousVisibleIniLoad: {
-              on: 'ini-load',
-              request: 'iniLoadCsi',
-              selector: 'amp-ad',
-              selectionMethod: 'closest',
-            },
-            continuousVisibleRenderStart: {
-              on: 'render-start',
-              request: 'renderStartCsi',
-              selector: 'amp-ad',
-              selectionMethod: 'closest',
-            },
-          },
-        });
+        expect(config).to.deep.equal(builtConfig);
         headers.has = function(name) {
           expect(name).to.equal('X-AmpAnalytics');
           return false;
         };
         expect(extractAmpAnalyticsConfig(a4a, headers)).to.not.be.ok;
       });
     });
+
+    it('should add the correct CSI signals', () => {
+      const mockA4a = {
+        win: window,
+        element: {
+          getAttribute: function(name) {
+            switch (name) {
+              case EXPERIMENT_ATTRIBUTE:
+                return '00000001,00000002';
+              case 'type':
+                return 'fake-type';
+              case 'data-amp-slot-index':
+                return '0';
+            }
+          },
+        },
+      };
+      const headers = {
+        get: function(name) {
+          if (name == 'X-QQID') {
+            return 'qqid_string';
+          }
+        },
+        has: function(name) {
+          if (name == 'X-QQID') {
+            return true;
+          }
+        },
+      };
+      let newConfig = addCsiSignalsToAmpAnalyticsConfig(
+          mockA4a, builtConfig, headers,
+          true /* indicates that this is rendered via friendly frame */,
```

Comment:
style nit: I believe it is common practice to put comments like this before the parameter, instead of after.